spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/ProducerProperties.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder;
+
+import org.springframework.expression.Expression;
+
+/**
+ * Common producer properties.
+ *
+ * @author Marius Bogoevici
+ */
+public class ProducerProperties {
+
+	private Expression partitionKeyExpression = null;
+
+	private Class<?> partitionKeyExtractorClass = null;
+
+	private Class<?> partitionSelectorClass = null;
+
+	private Expression partitionSelectorExpression = null;
+
+	private boolean batchingEnabled = false;
+
+	private int batchSize;
```

Comment:
Actually @ilayaperumalg points out that they're used only by Rabbit. There we go.